packages/backend/src/payment-method/ilp/quote-details/service.ts

```diff
@@ -0,0 +1,32 @@
+import { TransactionOrKnex } from 'objection'
+import { IlpQuoteDetails } from './model'
+import { BaseService } from '../../../shared/baseService'
+
+export interface IlpQuoteDetailsService {
+  getByQuoteId(quoteId: string): Promise<IlpQuoteDetails | undefined>
+}
+
+export interface ServiceDependencies extends BaseService {
+  knex: TransactionOrKnex
+}
+
+export async function createIlpQuoteDetailsService(
+  deps_: ServiceDependencies
+): Promise<IlpQuoteDetailsService> {
+  const deps = {
+    ...deps_,
+    logger: deps_.logger.child({ service: 'IlpQuoteDetailsService' })
+  }
+  return {
+    getByQuoteId: (quoteId: string) =>
+      getIlpQuoteDetailsByQuoteId(deps, quoteId)
+  }
+}
```

Comment:
related: https://github.com/interledger/rafiki/pull/2857/files#r1767027492

I created an entirely new service for this table. Initially I though maybe I could just put it in the quote service but it never felt quite right (quote service returning record from not the quotes table, concerned with ILP specific payment method stuff, etc.). Additionally this created a circular dependency when requiring quote service in the payment method - so I think we have to factor out like this either way.

Reply:
I think I'm coming around to that. 

My immediate reaction was that I'd like to keep the objection calls in the service layers so it can more easily be used across other services and unit tested. And general seperations of concern. Although I recognize the unit test is trivial (just gets by id) and would be implicitly tested by the payment tests. And it's pretty likely we only need the ilp details here. 

Additionally, we are already interfacing with the `ilpQuoteDetails` outside the `ilpQuoteDetail` service in `quoteService.create` (its a bit abstracted away via the objection api but its happening in the `insertGraphAndFetch`). If I wanted to really enforce going through the service layer I guess that would need to change to use some `ilpQuoteDetailsService.create`. And it does generally feel a bit weird to treat it as a thing in-and-of itself (rather than just as _maybe_ part of the quote).